src/eytzinger_benchmark.zig

```diff
@@ -0,0 +1,319 @@
+const std = @import("std");
+const assert = std.debug.assert;
+const math = std.math;
+
+const binary_search = @import("./binary_search.zig").binary_search;
+const eytzinger = @import("./eytzinger.zig").eytzinger;
+const perf = @import("./benchmarks/perf.zig");
+
+const GiB = 1 << 30;
+const searches = 100_000;
+
+const kv_types = .{
+    .{.key_size = 8, .value_size = 128},
+    .{.key_size = 8, .value_size = 64},
+    .{.key_size = 16, .value_size = 16},
+    .{.key_size = 32, .value_size = 32},
+};
+
+// keys_per_summary = values_per_page / summary_fraction
+const summary_fractions = .{4, 8, 16, 32};
+const values_per_page = .{128, 256, 512, 1024, 2048, 4096, 8192};
+const body_fmt = "{:_>2}B/{:_>3}B {:_>4}/{:_>4} {s}{s}: WT={:_>6}ns UT={:_>6}ns CY={:_>6} IN={:_>6} CR={:_>5} CM={:_>5} BM={}\n";
+
+const summary_sizes = blk: {
+    var sizes: [values_per_page.len][summary_fractions.len]usize = undefined;
+    for (values_per_page) |values_count, v| {
+        for (summary_fractions) |fraction, k| {
+            // Set in reverse order so that the summary sizes ascend.
+            sizes[v][summary_fractions.len - k - 1] = values_count / fraction;
+        }
+    }
+    break :blk sizes;
+};
+
+pub fn main() !void {
+    std.log.info("Samples: {}", .{searches});
+    std.log.info("WT: Wall time/search", .{});
+    std.log.info("UT: utime time/search", .{});
+    std.log.info("CY: CPU cycles/search", .{});
+    std.log.info("IN: instructions/search", .{});
+    std.log.info("CR: cache references/search", .{});
+    std.log.info("CM: cache misses/search", .{});
+    std.log.info("BM: branch misses/search", .{});
+
+    var seed: u64 = undefined;
+    try std.os.getrandom(std.mem.asBytes(&seed));
+    var prng = std.rand.DefaultPrng.init(seed);
+
+    // Allocate from heap just once. All page allocations reuse this buffer.
+    var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
+    defer arena.deinit();
```

Comment:
```suggestion
    defer arena.deinit();
    
```

We leave a newline after `defer` and `errdefer` and the last `assert()` in a block of assertions because these are critical sections and that helps them stand out. You can then immediately see that the allocation is balanced with a de-allocation, to make leaks easier to spot.